added tabIndex to the links

caperneoignis opened the pull request:
This should resolve the issue of tab accessibility, but I will manually test to make sure

caperneoignis committed:
added tabIndex to the links

This should resolve the issue of tab accessibility, but I will manually test to make sure

caperneoignis committed:
corrected formating to resolve error

corrected formating to resolve error.

caperneoignis committed:
updated with aria-label even though may not need it

Currently trying to figure out how to make the table accessible, got links working which at least makes the page usuable.

caperneoignis committed:
corrected error

can't use underscores...

caperneoignis committed:
needed to correct indent

caperneoignis committed:
one more indent fix

one more indent fix.

garemoko commented:
Thanks for this. Couple of items from a very quick initital look:

1. You're deleting the Brazilian Portuguese language file!
2. There seem to be a lot of whitespace changes - are those needed to pass the tests? Ideally, please submit the whitespace changes as a separate PR first (which i'll merge quickly) to make the actual code changes easier to review.

caperneoignis committed:
updated to have better key detection and screen reader functionality

Still working on this, should be done with a bit more testing.

caperneoignis commented:
@garemoko the ci tool is throwing errors for some of the white space issues. I'll look to see what exactly is being changed. It could be my IDE is doing cleanup before submitting. Also I replied to your other comment in the other commit. The copy and paste detection is throwing an error about the Brazilian version since there is no difference between the two. Moodle will default to the main language file which in this case is pr, if pr_br is not present, in this case since both arrays and strings are the same, you will not see any differences if you are on a pr_br language setting while in Moodle, so we don't need the file, unless you are planning on adding a change later on, in which case you only need to add that one array element with the different string, other wise moodle will default to PR base language file. But I'm happy to add it back, the CI will just fail the jobs.

caperneoignis committed:
updated to correct lint error

Lines were too long so corrected that issue, and updated string concation for consistanticy.

caperneoignis committed:
correcting lint issues

Corrected two of the lint issues.

caperneoignis committed:
updated link location and converted tabs to spaces.

Tabs found there way in to the file, updated to spaces.

caperneoignis committed:
white space removal

garemoko commented:
@caperneoignis are you sure the lang files are 100% identical? If so, again I'd prefer to have that in a separate PR to keep the history clean.

caperneoignis commented:
@garemoko okay, I'll add it back, and we can make another pull request. I checked they are 100% identical.

caperneoignis committed:
add Brizillian language file back

caperneoignis commented:
@garemoko done, both branches have the language file back.

caperneoignis commented:
@garemoko hold on doing this merge right now, I'm waiting on my 508 person to do a scan, and she won't be back till Friday. However, I think this will pass accessibility, but I may need to tweek it based on her feed back.

garemoko commented:
Also need somebody else to test this

caperneoignis commented:
@garemoko Hey, just FYI this is good to go from my end. If anyone tests it and finds and issue I'll update. Let me know if you find any thing. Thanks.

garemoko commented on view.php:
```diff
@@ -116,23 +122,18 @@ function mod_tincanlaunch_launchexperience(registration) {
-            ."')\" style=\"cursor: pointer;\">"
-            .get_string('tincanlaunch_attempt', 'tincanlaunch')
-            ."</a></p>";
-    }
```
Why are you removing this section?

garemoko replied:
Oh I see. You moved it. How does this affect the UI? If it's different, can you share a screenshot?

caperneoignis replied:
It's only different if you have previous attempts. Otherwise it shows up like it use to. Reason being, is because we don't have a warning or any text with the 'launch new attempt' so a blind user will only hear 'launch attempt' and not hear/know about the table with their previous attempts. This way, the non-sighted user will be able to hear about the previous attempts before hearing launch new attempt.

caperneoignis replied:
The blind person can't see the layout, so we have to order it so we give them previous attempts first before they hear launch new. Otherwise they may think it is the only link on the page, or that link is the one that will launch the previous attempt.

caperneoignis replied:
![after attempt](https://user-images.githubusercontent.com/3696698/41057784-07a4c7e8-6996-11e8-9065-fdab1b60a219.png)


FYI, I have to truncate some of the information out, due to the fact I'm working on another project with the plugin. So sorry in advance for a super small glimpse of the page.

caperneoignis replied:
If you haven't done an attempt or it's the return home link. It will look the same as before.

garemoko commented on view.php:
```diff
@@ -145,28 +146,40 @@ function mod_tincanlaunch_launchexperience(registration) {
     );
     $table->data = $registrationdatafromlrs;
     echo html_writer::table($table);
+    // Needs to come after previous attemps so a non-sighted user can hear launch options.
```
missing t in comment

caperneoignis replied:
Gotcha, I'll update.

caperneoignis committed:
corrected misspell in comment.

garemoko commented:
I'll test this as I work on the next release.
